src/System.Diagnostics.Process/src/System/Diagnostics/Process.FreeBSD.cs

```diff
@@ -64,8 +64,16 @@ public TimeSpan PrivilegedProcessorTime
         }
 
         /// <summary>Gets parent process ID</summary>
-        private int ParentProcessId =>
-            throw new PlatformNotSupportedException();
+        private unsafe int ParentProcessId
+        {
+            get
+            {
+                EnsureState(State.HaveNonExitedId);
+
+                Interop.Process.kinfo_proc* stat = Interop.Process.GetProcInfo(_processId, false, out _);
+                return stat->ki_ppid;
```

Comment:
You need to check whether stat is null. I think you also sould check the returned count.

See the other uses in this file for examples.

Reply:
Added a check, and renamed `stat` to `processInfo` to align with exception SR name.